client/components/quizSelection/TopicSelection.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { TopicSelectionContainer, TopicSelectionList, TopicSelectionItem } from './styles';
+import { PrimaryButton, PrimaryButtonSmall, TextBodySmall } from '../shared/styles';
+
+export default function TopicSelection({ subject, setChosenSubject, chosenTopics, setChosenTopics }) {
+
+  const toggleTopics = (topic) => {
+    if (chosenTopics.includes(topic)) {
+      setChosenTopics(topics => topics.filter(t => t !== topic));
+    } else {
+      setChosenTopics(topics => [...topics, topic]);
+    }
+  };
+
+  return (
+    <TopicSelectionContainer>
+      <TopicSelectionList>
+        <TopicSelectionItem>
+          <PrimaryButton mod={subject === "Programming" ? "fillLight" : "ghost"} onClick={() => {setChosenSubject("Programming"); setChosenTopics([])}}>
+            <TextBodySmall>Programming</TextBodySmall>
+          </PrimaryButton>
+        </TopicSelectionItem>
+        <TopicSelectionItem>
+          <PrimaryButton mod={subject === "UX" ? "fillLight" : "ghost"} onClick={() => {setChosenSubject("UX"); setChosenTopics([])}}>
```

Comment:
Calling multiple functions inside an onclick looks like bad practice.
In this case, it should probably have a local onClick function (ex. "onSetSubject"), which then calls functions

Reply:
Great point!

Reply:
Fixed in latest push, made an onClick handler for this